### PR TITLE
net-misc/onedrive: add 2.5.3

### PR DESCRIPTION
**Pull request by douglarek**
I have been using OneDrive version 2.5 for a long time, and now version 2.5.3 is very stable. I recommend checking in.

### CODE REVIEW

**Review by the-horo — CHANGES_REQUESTED** [review]
Thank you for contributing.

I only have one minor nit with the ebuild but onedrive-2.5 seems to be working fine otherwise. @MrDuartePT is the maintainer, however, so let's see what he has to say.

File: net-misc/onedrive/onedrive-2.5.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	# init script
+	dobin contrib/init.d/onedrive_service.sh
+	newinitd contrib/init.d/onedrive.init onedrive
+}
+
+pkg_postinst() {
+	elog "OneDrive Free Client needs to be authorized to access your data before the"
+	elog "first use. To do so, run onedrive in a terminal for the user in question and"
+	elog "follow the steps on screen."
+}
```
Comment (the-horo) [inline net-misc/onedrive/onedrive-2.5.3.ebuild:59]: Upstream mentions that the 2.5 releases are not backward-compatible with 2.4 and some care needs to be taken like making sure you update all your devices to 2.5. I think the user should be warned about this.

```suggestion

	local old_ver minor_part="$(ver_cut 1-2)"
	for old_ver in "${REPLACING_VERSIONS[@]}"; do
		if ver_test "${old_ver}" -lt "${minor_part}"; then
			ewarn "You are performing an upgrade that is not backwards-compatible"
			ewarn "and you need to upgrade to ${PN}-${minor_part} on all your devices."
			ewarn "Please read: https://github.com/abraunegg/onedrive/releases/tag/v${PV}"
			break
		fi
	done
}
```

**Comment by the-horo** [comment]
Can you squash the second commit so we would be left with a single one? It:
1. keeps the history cleaner
2. preserves the `Signed-off-by: `. As per the [Gentoo copyright policy](https://devmanual.gentoo.org/general-concepts/copyright-policy/index.html#certificate-of-origin), each commit must be `--signoff`'ed in order to be merged.

**Review by the-horo — APPROVED** [review]

**Comment by MrDuartePT** [comment]
> Can you squash the second commit so we would be left with a single one? It:
> 
> 1. keeps the history cleaner
> 
> 2. preserves the `Signed-off-by: `. As per the [Gentoo copyright policy](https://devmanual.gentoo.org/general-concepts/copyright-policy/index.html#certificate-of-origin), each commit must be `--signoff`'ed in order to be merged.

Everthing look ok I didn't update the ebuild when that version release because of the warning that is not backwards compatible with 2.4.x

https://github.com/abraunegg/onedrive/releases/tag/v2.5.3

Thanks for the PR, but I think it better to mask this version until some user test it.

**Comment by MrDuartePT** [comment]
This program is mission critical for me since I use it to sync my uni OneDrive that why I didn't want to risk breaking something.

**Comment by the-horo** [comment]
I think it's fine to merge this PR with `KEYWORDS=''` and then add `~amd64` after some time, corresponding to upstream releases becoming more stable to avoid big bugs like in https://github.com/abraunegg/onedrive/releases/tag/v2.5.1. This should help with users testing the package while avoiding issues for people who depend on the package working. @MrDuartePT does this sound reasonable?

**Comment by MrDuartePT** [comment]
> @MrDuartePT does this sound reasonable?

Yes it sound reasonable to me.

**Pull request closed without merging**